src/Cake.Common/EnviromentAliases.cs

```diff
@@ -186,6 +186,58 @@ public static bool IsRunningOnUnix(this ICakeContext context)
             return context.Environment.Platform.IsUnix();
         }
 
+        /// <summary>
+        /// Determines whether the build script running on a macOS based system.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// if (IsRunningOnMac())
+        /// {
+        ///     Information("macOS!");
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="context">The context.</param>
+        /// <returns>
+        ///   <c>true</c> if the build script running on a macOS based system; otherwise <c>false</c>.
+        /// </returns>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Platform")]
+        public static bool IsRunningOnMac(this ICakeContext context)
```

Comment:
I went for "Mac" here as this is what seems to be used more commonly (and maybe ages better since OS "X" is "10" and maybe macOS finally decides to go to 11 in 50 years)

Also, the basic search of people implementing their own version is 3K "Mac" and 94 "OSX" (with various casing)

Reply:
Perhaps `IsRunningOnMacOs` is a better name here? What do you think @cake-build/cake-team

Reply:
I left it off because the casing was weird 😬 but there are lots of ways to write that...

Reply:
I also think that `IsRunningOnMacOs` is more appropriate. The term "Mac" refers to the line of computers (which comprises products like MacBook, Mac Pro and Mac mini), while macOS is the operating system that runs on those computers.

@mattleibow I think you're right in avoiding any reference to 'X' or "10", since the next version of [macOS (Big Sur)](https://en.wikipedia.org/wiki/MacOS_Big_Sur) coming out this fall will mark the move to 11. 🎉

Reply:
Agree with @patriksvensson & @ecampidoglio 👍